architecture/transaction-layer.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Transaction Layer
+summary: 
+toc: false
+---
+
+CockroachDB aims to provide transactions with the same kind of simplicity you would experience if you developed against a local MySQL instance, despite allowing massively distributed deployments.
+
+<div id="toc"></div>
+
+## Overview
+
+To provide consistency, CockroachDB implements full support for ACID transaction semantics in the Transaction Layer.
+
+However, it's important to realize that *all* statements are handled as transactions, including single statements––this is sometimes referred to as "autocommit mode" because it behaves as if every statement is followed by a `COMMIT`. If you want to treat multiple statements as a single atomic unit, you can use the SQL standard `BEGIN`/`COMMIT`/`ROLLBACK` syntax.
+
+Transactional consistency is accomplished through a two-phase commit process.
+
+### Writing & Checking Intents (Phase 1)
+
+Instead of accessing values directly, CockroachDB instead handles access to MVCC values by writing two things:
+
+- A **Transaction Record** stored in the range where the first write occurs. This record includes the transaction’s current state (starting as `PENDING`), as well as a randomly assigned priority, which will be used to handle some cases of contention with other transaction. Though it’s random, the client can influence the priority by explicitly setting it to `LOW` or `HIGH` when initiating the transaction.
+
+- **Intents* for all of a transaction’s writes, which represent a provisional, uncommitted state. These are essentially the same as standard multi-version concurrency control (MVCC) values but also contain a special intent flag that points to the Transaction Record stored on the cluster. As long as the value exists as an intent, its values are treated as provisional.
+
+   As intents are written, CockroachDB checks for existing intents for the same keys. If it encounters any, it checks its transaction record; if the transaction is `PENDING`, the transaction waits for the existing one to resolve; otherwise it resolves `COMMITTED` values or removes `ABORTED` ones.
```

Comment:
what does it mean to "resolve" a committed value?

Reply:
Reworded slightly.